### PR TITLE
[#476] - db click navigation

### PR DESCRIPTION
**Pull request by Kasmadei**
Fix for: https://github.com/kbss-cvut/fta-fmea-ui/issues/476

### CODE REVIEW

File: src/components/editor/faultTree/Editor.tsx [hunk 1]
```diff
@@ -206,25 +206,31 @@ const Editor = () => {
...
   const redirectToInstance = (navigateFrom: string) => {
     if (faultTree?.manifestingEvent?.children) {
-      let map: { [name: string]: string } = {};
-      let children = faultTree?.manifestingEvent?.children;
-      if (faultTree?.manifestingEvent?.children.length > 0) {
-        faultTree?.manifestingEvent?.children.map((child) => {
-          if (child.children) {
-            children = [...children, ...child.children];
+      // Recursive function to traverse the tree and find the desired key-value pair
+      const findNodeValue = (children) => {
```
Comment (blcham) [inline src/components/editor/faultTree/Editor.tsx:210]: @Kasmadei I mentioned at least twice that you should not implement a recursive traverse function but instead save the state in the FTA node. I even described it here:
https://github.com/kbss-cvut/fta-fmea-ui/issues/476#issuecomment-2195609473

So why you needed to implement recursive function here?

**Review by blcham — CHANGES_REQUESTED** [review]
Explain or change

File: src/components/editor/faultTree/Editor.tsx (hunk 1 above)
Comment (Kasmadei) [reply src/components/editor/faultTree/Editor.tsx]: Refactored. I found a way how to refactor this, so traverse function removed.

**Review by blcham — APPROVED** [review]
LGTM

**Pull request merged**